app/src/main/java/com/dcrandroid/dialog/send/SendDialog.kt

```diff
@@ -125,9 +128,58 @@ class SendDialog(val fragmentActivity: FragmentActivity, dismissListener: Dialog
         clearEstimates()
     }
 
+    override fun onPause() {
+        super.onPause()
+        savedInstanceState = Bundle()
+        // Included this boolean value to avoid insufficient balance error if true onResume.
+        savedInstanceState?.putBoolean(Constants.SEND_MAX, sendMax)
+
+        // Fetch and save the selected source account.
+        val selectedSourceAccount = amountHelper.selectedAccount
+        savedInstanceState?.putSerializable(Constants.SELECTED_SOURCE_ACCOUNT, selectedSourceAccount)
+
+        // Fetch and save the selected destination account.
+        val selectedDestAccount = destinationAddressCard.destinationAccountSpinner.selectedAccount
+        savedInstanceState?.putSerializable(Constants.SELECTED_DESTINATION_ACCOUNT, selectedDestAccount)
+
+        // Save destination address state depending on if spinner or address input was selected.
+        if (destinationAddressCard.isSendToAccount) {
+            savedInstanceState?.putBoolean(Constants.SEND_TO_ACCOUNT, true)
+        } else {
+            savedInstanceState?.putBoolean(Constants.SEND_TO_ACCOUNT, false)
+        }
+    }
+
     override fun onResume() {
         super.onResume()
         destinationAddressCard.addressInputHelper.onResume()
+        if (savedInstanceState != null) {
+            // Fetch saved sendMax Value.
+            sendMax = savedInstanceState!!.getBoolean(Constants.SEND_MAX)
+
+            // Fetch saved destination address state.
+            val sendToAccount = savedInstanceState!!.getBoolean(Constants.SEND_TO_ACCOUNT)
+
+            // Fetch saved source and destination accounts.
+            val selectedSourceAccount = savedInstanceState!!.getSerializable(Constants.SELECTED_SOURCE_ACCOUNT) as Account?
+            val selectedDestAccount = savedInstanceState!!.getSerializable(Constants.SELECTED_DESTINATION_ACCOUNT) as Account
+
+            // Update UI based on previously selected source account.
+            sourceAccountSpinner.selectedAccount = selectedSourceAccount
+
+            // Update UI based on previously selected send to account.
+            destinationAddressCard.destinationAccountSpinner.selectedAccount = selectedDestAccount
+
+            // Show destination address input / spinner depending on which was previously selected.
+            if (sendToAccount) {
+                destinationAddressCard.addressInputHelper.hide()
+                destinationAddressCard.destinationAccountSpinner.show()
+            } else {
+                destinationAddressCard.addressInputHelper.show()
+                destinationAddressCard.destinationAccountSpinner.hide()
+            }
```

Comment:
No need to declare a variable outside if it's only used once.
```suggestion
            val selectedSourceAccount = savedInstanceState!!.getSerializable(Constants.SELECTED_SOURCE_ACCOUNT) as Account?
            sourceAccountSpinner.selectedAccount = selectedSourceAccount
            
            val selectedDestAccount = savedInstanceState!!.getSerializable(Constants.SELECTED_DESTINATION_ACCOUNT) as Account
           destinationAddressCard.destinationAccountSpinner.selectedAccount = selectedDestAccount

            // Show destination address input / spinner depending on which was previously selected.
            val sendToAccount = savedInstanceState!!.getBoolean(Constants.SEND_TO_ACCOUNT)
            if (sendToAccount) {
                destinationAddressCard.addressInputHelper.hide()
                destinationAddressCard.destinationAccountSpinner.show()
            } else {
                destinationAddressCard.addressInputHelper.show()
                destinationAddressCard.destinationAccountSpinner.hide()
            }
```